flocker/testtools/__init__.py

```diff
@@ -868,6 +870,76 @@ def __str__(self):
         return base + " and output:\n" + lines
 
 
+class _LoggingProcessProtocol(ProcessProtocol):
+    """
+    A ``ProcessProtocol`` that stores output from the subprocess and also logs
+    output as its received as eliot messages.
```

Comment:
"...that both stores and logs output from the subprocess. Output is logged as it is received." :question: